app/models/candidates/verification_code.rb

```diff
@@ -2,23 +2,37 @@ class Candidates::VerificationCode
   include ActiveModel::Model
   include ActiveModel::Attributes
   include ActiveModel::Validations::Callbacks
+  include ActiveRecord::AttributeAssignment
 
-  attribute :code
+  attribute :code, :string
+  # Will be pre-validated by another model.
+  attribute :email, :string
+  attribute :firstname, :string
+  attribute :lastname, :string
+  attribute :date_of_birth, :date
 
   validates :code, length: { is: 6 }, format: { with: /\A[0-9]*\z/ }
 
   before_validation if: :code do
     self.code = code.to_s.strip
   end
 
-  def exchange(personal_information)
+  def issue_verification_code
```

Comment:
Would it be better to split this method to a command and a query method?

```
def issue_verification_code
  request = GetIntoTeachingApiClient::ExistingCandidateRequest.new(matchback_attributes)
  GetIntoTeachingApiClient::CandidatesApi.new.create_candidate_access_token(request)
end

def verification_code_issued?
  issue_verification_code
  true
rescue GetIntoTeachingApiClient::ApiError
  false
end

```
or if the create_candidate_access_token returns something, then
```
def verification_code_issued?
  issue_verification_code
rescue GetIntoTeachingApiClient::ApiError
  false
end
```

Reply:
I'm on the fence with this one - I think there's a better way of structuring it, but to me having `verification_code_issued?` triggering the action feels misleading with respect to the method naming. I'd prefer to leave this bit as-is for now and rework it later on. I think the API could do a better job here and return a status instead of raising a `404` when the candidate wasn't found, which would help clear up the app code too.